src/Point.ts

```diff
@@ -0,0 +1,35 @@
+export class Point {
+  constructor(private x: number = 0, private y: number = 0) {
+    this.x = x;
+    this.y = y;
+  }
+
+  distance(x?: number | Point, y?: number): number {
```

Comment:
Same for distance method - we need to have overloaded method as for task.

Reply:
Method overried applied